src/variogram.rs

```diff
@@ -192,29 +184,34 @@ pub fn variogram_ma_structured(
     fn inner<E: Estimator>(f: ArrayView2<'_, f64>, mask: ArrayView2<'_, bool>) -> Array1<f64> {
         let size = f.dim().0;
 
-        let mut variogram = Array1::<f64>::zeros(size);
-        let mut counts = Array1::<u64>::zeros(size);
+        let mut variogram = Vec::new();
 
-        Zip::indexed(variogram.slice_mut(s![1..]))
-            .and(counts.slice_mut(s![1..]))
-            .par_for_each(|k, variogram, counts| {
-                Zip::from(f.slice(s![..size - k - 1, ..]))
-                    .and(f.slice(s![k + 1.., ..]))
-                    .and(mask.slice(s![..size - k - 1, ..]))
-                    .and(mask.slice(s![k + 1.., ..]))
+        (0..size)
```

Comment:
Just noticed one thing: This now computes the case `k=0` which is pointless as it will always result in zero and was not computed before this change. I could not keep that as `collect_into_vec` always truncates, but I could probably use `par_extend` instead of just push zero into the vector as the first element...

Reply:
Added a commit making the switch to `par_extend`. Also took the library of adding another commit updating the locked dependencies.